### DISCUSSION TITLE
is it ok to exposed the firebase config to the client?

### DISCUSSION THREAD

**Discussion by jozhuare99** [firebase/firebase-js-sdk#8163] [Q&A]
![Screenshot from 2024-04-13 09-04-32](https://github.com/firebase/firebase-js-sdk/assets/47843347/efbdf9b3-aa68-4c23-b3fb-5c1c833db515)
is it fine to be exposed the config?

**Comment by JEROME-PRAKASH-L** [comment]
Yes—the Firebase **web config object is designed to be included in client code**. Firebase documents its values (including the Firebase-provisioned API key, project ID, app ID, database URL, and storage bucket) as non-secret identifiers.

That does **not** mean the backend is automatically public. Protect it with:

- Firebase Security Rules for Firestore, Realtime Database, and Storage
- Firebase Authentication where access should be user-specific
- App Check where supported
- API-key restrictions limited to the Firebase APIs and app origins you actually use

Do not put genuinely privileged credentials in client code: service-account private keys, FCM server keys, Admin SDK credentials, or API keys for unrelated services such as the Gemini Developer API.

Official references: [Firebase config object](https://firebase.google.com/docs/web/learn-more#config-object), [API keys for Firebase](https://firebase.google.com/docs/projects/api-keys), and the [Firebase security checklist](https://firebase.google.com/support/guides/security-checklist).
